contracts/mocks/ReentrancyMock.sol

```diff
@@ -35,6 +35,8 @@ contract ReentrancyMock is ReentrancyGuard {
         _count();
         bytes4 func = bytes4(keccak256("callback()"));
         attacker.callSender(func);
+
+        require(_reentrancyGuardStatus() == 2, "ReentrancyMock: _reentrancyGuardStatus failed");
```

Comment:
I don't think this is getting executed, the attacker line above probably reverts.

Please add a separate test specifically for the guard status, and for the various values it can have.

Reply:
+1 for specific testing

Reply:
Done. 

Sorry for late response, busy doing a bunch of other things lol:)